File: Java/3.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        char[] arr = s.toCharArray();
        int max = 0;
        int left = 0;
        int right = 0;
        Set<Character> known = new HashSet<>();
        while(right < arr.length){
            if(known.add(arr[right])){
                right++;
                max = Math.max(max,known.size());
            }
            else{
                known.remove(arr[left++]);
            }
        }
        return max;
    }
}

File: CPP/3.cpp
using namespace std;

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        set<char> known;
        int left = 0;
        int right = 0;
        int maxLength = 0;
        while(right < s.length()){
            if(known.insert(s.at(right)).second){
                right++;
                maxLength = fmax(maxLength,known.size());
            }
            else{
                known.erase(s.at(left++));
            }
        }
        return maxLength;
    }



};
